src/core/tracing/Window.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Override some bits of foam.core.Window to support more accurate tracing.
+foam.CLASS({
+  package: 'foam.core.tracing',
+  name: 'Window',
+  extends: 'foam.core.Window',
+
+  requires: [ 'foam.core.tracing.Notification' ],
+  exports: [ 'lastEvent_' ],
+
+  properties: [
+    {
+      name: 'lastEvent_',
+      value: null
+    }
+  ],
+
+  methods: [
+    function getAsyncListener_(state) {
+      var ctx = this;
+      var baseListener = this.SUPER(state);
+      return function asyncListener() {
+        var tps = foam.lookup('foam.core.FObject').TRACING_PUB_SUB;
```

Comment:
You can just require: [ 'FObject' ] (since it's in foam.core. you don't need to specify it), then refer to it as 'this.FObject'.